packages/sdk/src/apiClient/basicAuthProvider.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AxiosRequestConfig } from "axios";
+import { AuthProvider } from "./authProvider";
+import { ErrorWithCode, ErrorCode, ErrorMessage } from "../core/errors";
+import { formatString } from "../util/utils";
+
+/**
+ * Provider that handles Basic authentication
+ *
+ * @beta
+ */
+export class BasicAuthProvider implements AuthProvider {
+  private userName: string;
+  private password: string;
+
+  /**
+   *
+   * @param userName - Username used in basic auth
+   * @param password - Password used in basic auth
+   *
+   * @beta
+   */
+  constructor(userName: string, password: string) {
+    if (!userName) {
+      throw new ErrorWithCode(
+        formatString(ErrorMessage.ArgumentEmpty, "username"),
+        ErrorCode.InvalidParameter
+      );
+    }
+    if (!password) {
+      throw new ErrorWithCode(
+        formatString(ErrorMessage.ArgumentEmpty, "password"),
+        ErrorCode.InvalidParameter
+      );
+    }
+    this.userName = userName;
+    this.password = password;
+  }
+
+  /**
+   * Adds authentication info to http requests
+   *
+   * @param config - Contains all the request information and can be updated to include extra authentication info.
+   * Refer https://axios-http.com/docs/req_config for detailed document.
+   *
```

Comment:
let's add parameter type, what's returned, what error it will throw to the document. You can refer existing SDK implementation about how to add these info.

Reply:
Fixed.